swiss-army-knife/overlaytest.sh

```diff
@@ -1,19 +1,68 @@
 #!/bin/bash
-echo "=> Start network overlay test"
-  kubectl get pods -l name=overlaytest -o jsonpath='{range .items[*]}{@.metadata.name}{" "}{@.spec.nodeName}{"\n"}{end}' | sort -k2 |
-  while read spod shost
-    do kubectl get pods -l name=overlaytest -o jsonpath='{range .items[*]}{@.status.podIP}{" "}{@.spec.nodeName}{"\n"}{end}' | sort -k2 |
-    while read tip thost
-      do
-        if [[ ! $shost == $thost ]]
-        then
-          kubectl --request-timeout='10s' exec $spod -c overlaytest -- /bin/sh -c "ping -c2 $tip > /dev/null 2>&1"
-          RC=$?
-          if [ $RC -ne 0 ]
-            then echo FAIL: $spod on $shost cannot reach pod IP $tip on $thost
-            else echo $shost can reach $thost
-          fi
-        fi
-    done
+
+DNS_CHECK=false
+
+# Parse arguments
+while [[ $# -gt 0 ]]; do
+  case $1 in
+    --dns-check)
+      DNS_CHECK=true
+      shift
+      ;;
+    *)
+      echo "Unknown option: $1"
+      exit 1
+      ;;
+  esac
+done
+
+echo "=> Start network overlay, DNS, and API test"
+
+kubectl get pods -l name=overlaytest -o jsonpath='{range .items[*]}{@.metadata.name}{" "}{@.spec.nodeName}{" "}{@.status.podIP}{"\n"}{end}' | sort -k2 |
+while read spod shost sip
+do
+  echo "Testing pod $spod on node $shost with IP $sip"
+
+  # Overlay network test
+  echo "  => Testing overlay network connectivity"
+  kubectl get pods -l name=overlaytest -o jsonpath='{range .items[*]}{@.status.podIP}{" "}{@.spec.nodeName}{"\n"}{end}' | sort -k2 |
+  while read tip thost
+  do
+    if [[ ! $shost == $thost ]]; then
+      kubectl --request-timeout='10s' exec $spod -c overlaytest -- /bin/sh -c "ping -c2 $tip > /dev/null 2>&1"
```

Comment:
I found the `--request-timeout='10s'` flag seems to stop kubectl from using the SA token to auth, and it errors like kubectl doesn’t have a kubeconfig. Strange, only repro’s when running in a pod. If you also find this, we may need to run without the flag?